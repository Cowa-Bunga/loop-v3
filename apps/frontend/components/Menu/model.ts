export const menuModel = [
  { id: 1, icon: '', title: 'Dashboard', link: '/' },
  { id: 2, icon: '', title: 'All Orders', link: '/orders' },
  { id: 3, icon: '', title: 'Driver Details', link: '/' },
  { id: 4, icon: '', title: 'Import Orders', link: '/' },
  { id: 5, icon: '', title: 'Training Centre', link: '/' },
  { id: 6, icon: '', title: 'Support', link: '/' }
]

export const menuBottomModel = [
  { id: 7, icon: '', title: 'Client Admin', link: '/' },
  { id: 8, icon: '', title: 'Fleet Management', link: '/' },
  { id: 9, icon: '', title: 'Log Out', action: () => 'signOut' }
]
