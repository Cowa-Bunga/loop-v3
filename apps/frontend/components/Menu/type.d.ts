import { ReactComponent } from '../../index'
interface IMenu {
  id?: string
  title: string
  link?: string
  action?: () => void
  icon?: () => ReactComponent
  permissions?: EnumDbPermissionsScopes[]
  children?: IMenu[]
}

interface IMenuProps {
  open: boolean
  setOpen: (open: boolean) => void
}
