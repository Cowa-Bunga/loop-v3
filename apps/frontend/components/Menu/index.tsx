import { MenuList, MenuItem, Box, Drawer } from '@mui/material'
import HomeIcon from '@mui/icons-material/Home'
import { useRouter } from 'next/router'
import { menuBottomModel, menuModel } from './model'
import ui from './style'

export default function Menu({ open }: { open: boolean }) {
  const router = useRouter()

  const goto = (url: string) => {
    router.push(url)
  }

  return (
    <Drawer variant="temporary" anchor="left" hideBackdrop open={open}>
      <Box sx={ui.container}>
        <MenuList sx={ui.menuList}>
          {menuModel.map((item) => (
            <MenuItem
              key={item.id}
              sx={ui.menuItem}
              onClick={() => goto(item.link)}
            >
              <HomeIcon sx={ui.homeIcon} />
              {item.title}
            </MenuItem>
          ))}

          <Box sx={ui.menuBottom}>
            {menuBottomModel.map((item) => (
              <MenuItem
                key={item.id}
                sx={ui.menuItem}
                onClick={() => goto(item.link)}
              >
                <HomeIcon sx={ui.icon} />
                {item.title}
              </MenuItem>
            ))}
            <Box sx={ui.version}>version: 3.1.0</Box>
          </Box>
        </MenuList>
      </Box>
    </Drawer>
  )
}
