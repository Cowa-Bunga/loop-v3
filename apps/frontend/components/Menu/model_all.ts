import DashboardIcon from '@mui/icons-material/Dashboard'
import BusinessIcon from '@mui/icons-material/Business'
import LocalShippingIcon from '@mui/icons-material/LocalShipping'
import SendTimeExtensionIcon from '@mui/icons-material/SendTimeExtension'
import GradingIcon from '@mui/icons-material/Grading'
import ImportExportIcon from '@mui/icons-material/ImportExport'
import SupportAgentIcon from '@mui/icons-material/SupportAgent'
import LiveHelpIcon from '@mui/icons-material/LiveHelp'
import ModelTrainingIcon from '@mui/icons-material/ModelTraining'
import AssessmentIcon from '@mui/icons-material/Assessment'
import AppSettingsAltIcon from '@mui/icons-material/AppSettingsAlt'
// import PowerSettingsNewIcon from '@mui/icons-material/PowerSettingsNew'
import AdminPanelSettingsIcon from '@mui/icons-material/AdminPanelSettings'
import SensorOccupiedIcon from '@mui/icons-material/SensorOccupied'
import LoopIcon from '@mui/icons-material/Loop'
import PersonPinIcon from '@mui/icons-material/PersonPin'
import GroupsIcon from '@mui/icons-material/Groups'
import IntegrationInstructionsIcon from '@mui/icons-material/IntegrationInstructions'
import ApiIcon from '@mui/icons-material/Api'
import ElectricCarIcon from '@mui/icons-material/ElectricCar'
import DatasetLinkedIcon from '@mui/icons-material/DatasetLinked'
import { BookOnline } from '@mui/icons-material'

export const MODEL_MENU = ({ client_id }) =>
  [
    {
      id: 'dashboard',
      icon: DashboardIcon,
      title: 'Dashboard',
      link: '/dashboard',
      permissions: []
    },
    {
      id: 'fleet',
      icon: LocalShippingIcon,
      title: 'Fleet',
      permissions: [],
      children: [
        {
          id: 'fleet-admin',
          icon: AdminPanelSettingsIcon,
          title: 'Admin',
          link: `/dashboard`,
          permissions: []
        },
        {
          id: 'fleet-drivers',
          icon: SensorOccupiedIcon,
          title: 'Drivers',
          link: `/company/todo-id/driver/todo-driver-id/details`,
          permissions: []
        },
        {
          id: 'fleet-vehicle',
          icon: ElectricCarIcon,
          title: 'Vehicles',
          link: `/company/todo-id/fleet/todo-vehicle-id`,
          permissions: []
        }
      ]
    },
    {
      id: 'orders',
      icon: LoopIcon,
      title: 'Orders',
      permissions: [],
      children: [
        {
          id: 'create-orders',
          icon: SendTimeExtensionIcon,
          title: 'Create Order',
          link: `/company/${client_id}/orders/create`,
          permissions: []
        },
        {
          id: 'orders-all',
          icon: GradingIcon,
          title: 'All Orders',
          link: `/company/${client_id}/orders/view`,
          permissions: []
        },
        {
          id: 'orders-import',
          icon: ImportExportIcon,
          title: 'Import Orders',
          link: '/company/todo-id/orders/import',
          permissions: []
        }
      ]
    },
    {
      id: 'reports',
      icon: AssessmentIcon,
      title: 'Reports',
      link: '/company/todo-id/reports',
      permissions: [],
      children: [
        {
          id: 'all-orders-report',
          icon: AssessmentIcon,
          title: 'All Orders',
          link: '/company/todo-id/settings',
          permissions: []
        },
        {
          id: 'trip-report',
          icon: AssessmentIcon,
          title: 'Trip',
          link: '/company/todo-id/settings',
          permissions: []
        },
        {
          id: 'logbook-report',
          icon: AssessmentIcon,
          title: 'Log Book',
          link: '/company/todo-id/settings',
          permissions: []
        },
        // lookerstudio reports
        {
          id: 'looker-average-duration',
          icon: DatasetLinkedIcon,
          title: 'Average Durations',
          link: '/company/todo-id/reports',
          permissions: []
        },
        {
          id: 'looker-orders',
          icon: DatasetLinkedIcon,
          title: 'Orders Overview',
          link: '/company/todo-id/reports',
          permissions: []
        },
        {
          id: 'looker-delivery-times',
          icon: DatasetLinkedIcon,
          title: 'Delivery Times',
          link: '/company/todo-id/reports',
          permissions: []
        },
        {
          id: 'looker-sla-time-bands',
          icon: DatasetLinkedIcon,
          title: 'SLA Time Bands',
          link: '/company/todo-id/reports',
          permissions: []
        },
        {
          id: 'looker-distance-ranges',
          icon: DatasetLinkedIcon,
          title: 'Distance Ranges',
          link: '/company/todo-id/reports',
          permissions: []
        },
        {
          id: 'looker-driver-monitoring',
          icon: DatasetLinkedIcon,
          title: 'Driver Monitoring',
          link: '/company/todo-id/reports',
          permissions: []
        },
        {
          id: 'looker-driver-scheduling',
          icon: DatasetLinkedIcon,
          title: 'Driver Scheduling',
          link: '/company/todo-id/reports',
          permissions: []
        },
        {
          id: 'looker-locations',
          icon: DatasetLinkedIcon,
          title: 'Locations',
          link: '/company/todo-id/reports',
          permissions: []
        },
        {
          id: 'looker-exceptions',
          icon: DatasetLinkedIcon,
          title: 'Exceptions',
          link: '/company/todo-id/reports',
          permissions: []
        }
      ]
    },

    {
      id: 'settings',
      icon: AppSettingsAltIcon,
      title: 'Settings',
      permissions: [],
      children: [
        {
          id: 'settings-billing',
          icon: AppSettingsAltIcon,
          title: 'Account',
          link: '/company/todo-id/settings',
          permissions: []
        },
        {
          id: 'settings-clients',
          icon: PersonPinIcon,
          title: 'Clients',
          link: '/company/todo-id/settings',
          permissions: []
        },
        {
          id: 'settings-integration',
          icon: ApiIcon,
          title: 'Integrations',
          link: '/company/todo-id/settings',
          permissions: []
        },
        {
          id: 'settings-optimisations',
          icon: IntegrationInstructionsIcon,
          title: 'Optimisations',
          link: '/company/todo-id/settings',
          permissions: []
        },
        {
          id: 'settings-org',
          icon: BusinessIcon,
          title: 'Organisation',
          link: '/company/todo-id/settings',
          permissions: []
        },

        {
          id: 'settings-users',
          icon: GroupsIcon,
          title: 'Users',
          link: '/company/todo-id/settings',
          permissions: []
        }
      ]
    },
    {
      id: 'support',
      icon: SupportAgentIcon,
      title: 'Support',
      permissions: [],
      children: [
        {
          id: 'support-main',
          icon: LiveHelpIcon,
          title: 'Documentation',
          link: '/support/documentation',
          permissions: []
        },
        {
          id: 'training',
          icon: ModelTrainingIcon,
          title: 'Training Centre',
          link: '/support/training',
          permissions: []
        }
      ]
    },
    {
      id: 'tools',
      icon: ModelTrainingIcon,
      title: 'Tools',
      children: [
        {
          id: 'flow',
          icon: BookOnline,
          title: 'Flow Demo',
          link: '/tools/flow',
          permissions: []
        },
        // {
        //   id: 'api-dos',
        //   icon: BookOnline,
        //   title: 'DOS API',
        //   link: '/support/api-doc',
        //   permissions: []
        // },
        {
          id: 'api-driver',
          icon: BookOnline,
          title: 'API docs',
          link: '/support/api-doc',
          permissions: []
        }
      ]
    }
    // {
    //   id: 'signout',
    //   icon: PowerSettingsNewIcon,
    //   title: 'Sign out',
    //   action: () => 'signOut',
    //   permissions: []
    // }
  ] as unknown as IMenu[]
