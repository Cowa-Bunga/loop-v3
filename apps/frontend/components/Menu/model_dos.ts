import DashboardIcon from '@mui/icons-material/Dashboard'
import ImportExportIcon from '@mui/icons-material/ImportExport'
import SupportAgentIcon from '@mui/icons-material/SupportAgent'
import LiveHelpIcon from '@mui/icons-material/LiveHelp'
import ModelTrainingIcon from '@mui/icons-material/ModelTraining'
import AssessmentIcon from '@mui/icons-material/Assessment'
import PowerSettingsNewIcon from '@mui/icons-material/PowerSettingsNew'
import LoopIcon from '@mui/icons-material/Loop'
import { Person } from '@mui/icons-material'

export const MODEL_MENU = ({ client_id }) =>
  [
    {
      id: 'home',
      icon: DashboardIcon,
      title: 'Home',
      link: '/dashboard',
      permissions: []
    },
    {
      id: 'orders',
      icon: LoopIcon,
      title: 'Orders',
      permissions: [],
      children: [
        {
          id: 'all-orders',
          icon: LoopIcon,
          title: 'Overview',
          link: `/organisation/${client_id}/orders/view`,
          permissions: []
        },
        {
          id: 'import',
          icon: ImportExportIcon,
          title: 'Import New',
          link: `/organisation/${client_id}/reports`,
          permissions: []
        },
        {
          id: 'trip-report',
          icon: AssessmentIcon,
          title: 'Trip Report',
          link: `/organisation/${client_id}/fleet`,
          permissions: []
        }
      ]
    },
    {
      id: 'driver',
      icon: Person,
      title: 'Drivers',
      // tmp
      link: `/drivers/overview`,
      permissions: []
    },
    {
      id: 'messages',
      icon: Person,
      title: 'Message Centre',
      permissions: []
    },
    {
      id: 'reports',
      icon: PowerSettingsNewIcon,
      title: 'Reports',
      link: `/organisation/${client_id}/reports`,
      permissions: []
    },
    {
      id: 'routing',
      icon: LiveHelpIcon,
      title: 'Route Planning',
      link: '/dashboard',
      permissions: []
    },
    {
      id: 'learn',
      icon: ModelTrainingIcon,
      title: 'Learning Centre',
      link: '/support/training',
      permissions: []
    },
    {
      id: 'support',
      icon: SupportAgentIcon,
      title: 'Support',
      permissions: [],
      link: '/support/documentation'
    }
  ] as unknown as IMenu[]
