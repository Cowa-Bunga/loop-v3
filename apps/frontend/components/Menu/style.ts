import HomeIcon from '@mui/icons-material/Home'
const ui = {
  container: {
    width: '300px',
    height: '100vh',
    pt: '80px',
    backgroundColor: 'primary.main',
    backgroundImage:
      'url("https://www.loop.co.za/wp-content/uploads/2021/12/BGright.svg")',
    backgroundSize: 'cover',
    backgroundPosition: '0 0',
    color: 'white'
  },

  menuList: {
    width: '100%',
    height: '100%'
  },

  version: {
    fontSize: '8px',
    textAlign: 'center',
    margin: '40px 0 10px 0'
  },

  menuItem: {
    height: '50px'
  },

  homeIcon: {
    mr: 1,
    color: 'white'
  },

  homeIconDropdown: {
    mr: 1,
    color: 'white',
    ml: -3
  },

  homeIconInner: {
    ml: -2
  },

  menuBottom: {
    position: 'fixed',
    bottom: 0,
    width: 300
  },

  icon: {
    mx: 1
  },

  accordion: {
    background: 'transparent !important',
    color: 'white',
    width: '100%',
    m: 0,
    border: 0,
    boxShadow: 0,
    p: '0 0 0 8px',
    '& *': {
      m: 0,
      border: 0,
      boxShadow: 0,
      p: 0
    }
  }
}

export default ui
