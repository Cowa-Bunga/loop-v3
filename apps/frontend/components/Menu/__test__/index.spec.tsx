import React from 'react'
import { render } from '@testing-library/react'
import Menu from '../index'

jest.mock('next/router', () => ({
  useRouter: jest.fn()
}))

describe('Menu', () => {
  it('should render successfully', () => {
    const { baseElement } = render(<Menu open={true} setOpen={() => ''} />)
    expect(baseElement).toBeTruthy()
    expect(baseElement).toMatchSnapshot()
  })
})
