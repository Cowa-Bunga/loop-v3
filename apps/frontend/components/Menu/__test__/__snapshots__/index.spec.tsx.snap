// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu should render successfully 1`] = `
<body
  style="padding-right: 1024px; overflow: hidden;"
>
  <div
    aria-hidden="true"
  />
  <div
    class="MuiDrawer-root MuiDrawer-modal MuiModal-root css-1u2w381-MuiModal-root-MuiDrawer-root"
    role="presentation"
  >
    <div
      data-testid="sentinelStart"
      tabindex="0"
    />
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation16 MuiDrawer-paper MuiDrawer-paperAnchorLeft css-4t3x6l-MuiPaper-root-MuiDrawer-paper"
      tabindex="-1"
    >
      <div
        class="MuiBox-root css-1wifhf6"
      >
        <ul
          class="MuiList-root MuiList-padding css-17osibc-MuiList-root"
          role="menu"
          tabindex="-1"
        >
          <li
            class="MuiButtonBase-root MuiMenuItem-root MuiMenuItem-gutters MuiMenuItem-root MuiMenuItem-gutters css-vrs6a6-MuiButtonBase-root-MuiMenuItem-root"
            role="menuitem"
            tabindex="0"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1eayuh3-MuiSvgIcon-root"
              data-testid="HomeIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M10 20v-6h4v6h5v-8h3L12 3 2 12h3v8z"
              />
            </svg>
            Dashboard
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </li>
          <li
            class="MuiButtonBase-root MuiMenuItem-root MuiMenuItem-gutters MuiMenuItem-root MuiMenuItem-gutters css-vrs6a6-MuiButtonBase-root-MuiMenuItem-root"
            role="menuitem"
            tabindex="-1"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1eayuh3-MuiSvgIcon-root"
              data-testid="HomeIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M10 20v-6h4v6h5v-8h3L12 3 2 12h3v8z"
              />
            </svg>
            All Orders
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </li>
          <li
            class="MuiButtonBase-root MuiMenuItem-root MuiMenuItem-gutters MuiMenuItem-root MuiMenuItem-gutters css-vrs6a6-MuiButtonBase-root-MuiMenuItem-root"
            role="menuitem"
            tabindex="-1"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1eayuh3-MuiSvgIcon-root"
              data-testid="HomeIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M10 20v-6h4v6h5v-8h3L12 3 2 12h3v8z"
              />
            </svg>
            Driver Details
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </li>
          <li
            class="MuiButtonBase-root MuiMenuItem-root MuiMenuItem-gutters MuiMenuItem-root MuiMenuItem-gutters css-vrs6a6-MuiButtonBase-root-MuiMenuItem-root"
            role="menuitem"
            tabindex="-1"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1eayuh3-MuiSvgIcon-root"
              data-testid="HomeIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M10 20v-6h4v6h5v-8h3L12 3 2 12h3v8z"
              />
            </svg>
            Import Orders
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </li>
          <li
            class="MuiButtonBase-root MuiMenuItem-root MuiMenuItem-gutters MuiMenuItem-root MuiMenuItem-gutters css-vrs6a6-MuiButtonBase-root-MuiMenuItem-root"
            role="menuitem"
            tabindex="-1"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1eayuh3-MuiSvgIcon-root"
              data-testid="HomeIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M10 20v-6h4v6h5v-8h3L12 3 2 12h3v8z"
              />
            </svg>
            Training Centre
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </li>
          <li
            class="MuiButtonBase-root MuiMenuItem-root MuiMenuItem-gutters MuiMenuItem-root MuiMenuItem-gutters css-vrs6a6-MuiButtonBase-root-MuiMenuItem-root"
            role="menuitem"
            tabindex="-1"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1eayuh3-MuiSvgIcon-root"
              data-testid="HomeIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M10 20v-6h4v6h5v-8h3L12 3 2 12h3v8z"
              />
            </svg>
            Support
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </li>
          <div
            class="MuiBox-root css-13b9chb"
          >
            <li
              class="MuiButtonBase-root MuiMenuItem-root MuiMenuItem-gutters MuiMenuItem-root MuiMenuItem-gutters css-vrs6a6-MuiButtonBase-root-MuiMenuItem-root"
              role="menuitem"
              tabindex="-1"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1eayuh3-MuiSvgIcon-root"
                data-testid="HomeIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M10 20v-6h4v6h5v-8h3L12 3 2 12h3v8z"
                />
              </svg>
              Client Admin
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </li>
            <li
              class="MuiButtonBase-root MuiMenuItem-root MuiMenuItem-gutters MuiMenuItem-root MuiMenuItem-gutters css-vrs6a6-MuiButtonBase-root-MuiMenuItem-root"
              role="menuitem"
              tabindex="-1"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1eayuh3-MuiSvgIcon-root"
                data-testid="HomeIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M10 20v-6h4v6h5v-8h3L12 3 2 12h3v8z"
                />
              </svg>
              Fleet Management
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </li>
            <li
              class="MuiButtonBase-root MuiMenuItem-root MuiMenuItem-gutters MuiMenuItem-root MuiMenuItem-gutters css-vrs6a6-MuiButtonBase-root-MuiMenuItem-root"
              role="menuitem"
              tabindex="-1"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1eayuh3-MuiSvgIcon-root"
                data-testid="HomeIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M10 20v-6h4v6h5v-8h3L12 3 2 12h3v8z"
                />
              </svg>
              Log Out
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </li>
            <div
              class="MuiBox-root css-pz1d1y"
            >
              version: 3.1.0
            </div>
          </div>
        </ul>
      </div>
    </div>
    <div
      data-testid="sentinelEnd"
      tabindex="0"
    />
  </div>
</body>
`;
